sources/healthdog/src/service_check.rs

```diff
@@ -0,0 +1,127 @@
+use crate::error::{self, Result};
+use lazy_static::lazy_static;
+use log::trace;
+use regex::Regex;
+use snafu::ResultExt;
+use std::process::Command;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) struct ServiceHealth {
+    /// Whether or not the service is healthy.
+    pub(crate) is_healthy: bool,
+    /// In the event of an unhealthy service, the service's exit code (if found).
+    pub(crate) exit_code: Option<i32>,
+}
+
+pub(crate) trait ServiceCheck {
+    /// Checks the given service to see if it is healthy.
+    fn check(&self, service_name: &str) -> Result<ServiceHealth>;
+}
+
+pub(crate) struct SystemdCheck {}
+
+impl ServiceCheck for SystemdCheck {
+    fn check(&self, service_name: &str) -> Result<ServiceHealth> {
+        if is_ok(service_name)? {
+            return Ok(ServiceHealth {
+                is_healthy: true,
+                exit_code: None,
+            });
+        }
+        Ok(ServiceHealth {
+            is_healthy: false,
+            exit_code: parse_service_exit_code(service_name)?,
```

Comment:
If we can't parse it, should we return None rather than failing the whole check?  It's possible a systemd update breaks this module, and we'd want to know...

Reply:
Yeah good point, fixed in [2a5aef0](https://github.com/bottlerocket-os/bottlerocket/compare/f515d383c9e181c8a531a02414b320ff9231a81f..2a5aef04d7abcae5c91bf8cec84b8e95c7ccd306)